### ISSUE TITLE
Document that update intrinsics of C DB API cannot change primary key

### ISSUE THREAD

**Issue by arhag** [EOSIO/eos#4224]
The current (as of v1.0.5) update intrinsics of the C DB API do not allow changing the primary key of a table row. The serialized data that is stored in the table row of a primary table may include a primary key and that primary key value could be changed by the contract calling the `db_update_i64` intrinsic. But that does not change the actual primary key recognized by the native side.

This behavior should be well documented for contract developers that use the C DB API. Those using the C++ `multi_index` shouldn't have to worry about this (although it should still be documented) because the `modify` method will check that the primary key was not changed by the `updater` lambda and cause an assertion failure otherwise.

**Comment by dixia** [comment]
is this ticket still valid?

**Comment by lparisc** [comment]
Yes according to @arhag. The description or remarks section of the [db_update_i64()](https://github.com/EOSIO/eosio.cdt/blob/a6b8d3fc289d46f4612588cdd7223a3d549238f6/libraries/eosiolib/capi/eosio/db.h#L66) intrinsic just needs to specify that the primary key won't be updated (which is obvious but should still be said).
